cellxgene_schema_cli/cellxgene_schema/ontology_processing.py

```diff
@@ -0,0 +1,192 @@
+            if not onto.name == term_id.split(":")[0]:
+                continue
+
+            # If there are specified target terms then only work with them
+            if onto.name in owl_info:
+                if "only" in owl_info[onto.name]:
+                    if term_id not in owl_info[onto.name]["only"]:
+                        continue
+
+            onto_dict[onto.name][term_id] = dict()
+            try:
+                onto_dict[onto.name][term_id]["label"] = onto_class.label[0]
+            except IndexError:
+                onto_dict[onto.name][term_id]["label"] = ""
+
+            onto_dict[onto.name][term_id]["ancestors"] = [
+                i.name.replace("_", ":")
+                for i in onto_class.ancestors()
+                if i.name.split("_")[0] == onto.name
+            ]
+
+    with gzip.open(output_json_file, "wt") as output_json:
+        json.dump(onto_dict, output_json, indent=2)
+
+
+def _parse_gtf(gtf_path, output_file):
+    """
+    Parses a gziped GTF file to get gene and transcript info into a gziped comma-separated file with the following
+    structure, with three columns and no header: 1) gene/transcript id, 2) gene/transcript name,
+    3) gene/transcript version
+
+    :param str gtf_path: path to gzipped gtf file
+    :param str output_json_file: path to output json
+
+    :return: none
+    :rtype: none
+    """
+
+    output_to_print = ""
+    with gzip.open(gtf_path, "rb") as gtf:
+        for line in gtf:
+            line = line.decode("utf-8")
+
+            if line[0] == "#":
+                continue
+
+            line = line.rstrip().split("\t")
+
+            # Desired features based on whether is gene or transcript
+            if line[2] == "gene":
+                features = ["gene_id", "gene_name", "gene_version"]
+            elif line[2] == "transcript":
+                features = ["transcript_id", "transcript_name", "transcript_version"]
+            else:
+                continue
+
+            # Extract features (column 9 of GTF)
+            current_features = {
+                i.strip().split(" ")[0]: i.strip().split(" ")[1]
+                for i in line[8].split(";")
+                if len(i) > 1
+            }
+
+            # Select  features of interest, raise error feature of interest not found
+            target_features = [""] * len(features)
+            for i in range(len(features)):
+                feature = features[i]
+                if feature in current_features:
+                    current_features[feature] = current_features[feature].replace(
+                        '"', ""
+                    )
+                    target_features[i] = current_features[feature]
+
+            output_to_print += ",".join(target_features) + "\n"
+
+    with gzip.open(output_file, "wt") as output:
+        output.write(output_to_print)
+
+
+# Download and parse owls upon execution
+if __name__ == "__main__":
+    _dowload_owls()
```

Comment:
_dowload -> _download

Reply:
fixed in 4ac358decef7509c5944b70e93c9bd1e22eb45c1